digi/xbee/xsocket.py

```diff
@@ -0,0 +1,814 @@
+                lock.release()
+
+            # Check if the socket state has been received.
+            if not received_state:
+                raise TimeoutException("Timeout waiting for the socket connection")
+
+            # Check if the socket is connected successfully.
+            if received_state[0] != SocketState.CONNECTED:
+                raise XBeeSocketException(status=received_state[0])
+
+            self.__connected = True
+
+            # Register internal socket state and data reception callbacks.
+            self.__register_state_callback()
+            self.__register_data_received_callback()
+        finally:
+            # Always remove the socket state callback.
+            self.__xbee_device.del_socket_state_received_callback(socket_state_received_callback)
+
+    def bind(self, address):
+        """
+        Binds the socket to address. The socket must not already be bound.
+
+        Args:
+            address (Tuple): A pair ``(host, port)`` where ``host`` is the local interface (not used) and ``port`` is
+                the numeric port value.
+
+        Raises:
+            TimeoutException: if the bind response is not received in the configured timeout.
+            ValueError: if ``address`` is ``None`` or not a pair ``(host, port)``.
+            ValueError: if ``port`` is less than 1 or greater than 65535.
+            XBeeException: if the connection with the XBee device is not open.
+            XBeeSocketException: if the bind status is not ``SUCCESS``.
+            XBeeSocketException: if the socket is already bound.
+        """
+        # Check address and its contents.
+        if address is None or len(address) != 2:
+            raise ValueError("Invalid address, it must be a pair (host, port).")
+
+        port = address[1]
+        if port < 1 or port > 65535:
+            raise ValueError("Port number must be between 1 and 65535.")
+        if self.__source_port:
+            raise XBeeSocketException(status=SocketStatus.ALREADY_CONNECTED)
+
+        # If the socket is not created, create it first.
+        if self.__socket_id is None:
+            self.__create_socket()
+
+        # Create, send and check the socket create packet.
+        bind_packet = SocketBindListenPacket(self.__xbee_device.get_next_frame_id(), self.__socket_id, port)
+        response_packet = self.__xbee_device.send_packet_sync_and_get_response(bind_packet, self.__timeout)
+        self.__check_response(response_packet)
+
+        # Register the data reception from.
+        self.__register_data_received_from_callback()
+
+        # Store the source port.
+        self.__source_port = port
+
+    def listen(self, backlog=1):
+        """
+        Enables a server to accept connections.
+
+        Args:
+            backlog (Integer, optional): The number of unaccepted connections that the system will allow before refusing
+                new connections. If it's lower, it will be set to 0.
```

Comment:
What do you mean by if it's lower? Is it referring to the number of unaccepted connections being lower than the value specified, or vice-versa?